Server/src/models/GraphsModel.ts

```diff
@@ -93,6 +98,33 @@ export class GraphsModel {
         return sortedGraphData
     }
 
+    private async processGraphInput(graph: IGraphStateModel): Promise<any[]> {
+        let datasetIds: number[] = []
+        let datasetColors: string[] = []
+        let datasetShapes: string[] = []
+        let datasetHiddenStatus: boolean[] = []
+        let axisVariable: string[] = []
+        let axisLog: boolean[] = []
+        let axisZoomStart: number[] = []
+        let axisZoomEnd: number[] = []
+        let axisUnits: string[] = []
+        for (let j = 0; j < graph.datasets.length; j++) {
+            datasetIds.push(graph.datasets[j].id)
+            datasetColors.push(graph.datasets[j].color)
+            datasetShapes.push(graph.datasets[j].shape)
+            datasetHiddenStatus.push(graph.datasets[j].isHidden)
+        }
+        for (let k = 0; k < graph.axes.length; k++) {
+            axisVariable.push(graph.axes[k].variableName)
+            axisLog.push(graph.axes[k].logarithmic)
+            axisZoomStart.push(graph.axes[k].zoomStartIndex)
+            axisZoomEnd.push(graph.axes[k].zoomEndIndex)
+            axisUnits.push(graph.axes[k].units)
+        }
+        let processedData = [graph.id, graph.name, datasetIds, datasetColors, datasetShapes,

```

Comment:
Isnt this return a graphStateModel?

Reply:
No, it is not. The entire point of this method is to take a graphStateModel and change it into the format that it is stored in the database. That said, it could be refactored to pass a new object, say graphStateDatabase, which I will do.